packages/fiori/src/IllustratedMessage.js

```diff
@@ -386,15 +387,45 @@ class IllustratedMessage extends UI5Element {
 			return;
 		}
 
+		this._applyMedia();
+	}
+
+	_applyMedia() {
+		const bcr = this.getBoundingClientRect().toJSON();
+		let newMedia = "";
+
 		if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.BASE) {
-			this.media = IllustratedMessage.MEDIA.BASE;
+			newMedia = IllustratedMessage.MEDIA.BASE;
 		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.SPOT) {
-			this.media = IllustratedMessage.MEDIA.SPOT;
+			newMedia = IllustratedMessage.MEDIA.SPOT;
 		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.DIALOG) {
-			this.media = IllustratedMessage.MEDIA.DIALOG;
+			newMedia = IllustratedMessage.MEDIA.DIALOG;
 		} else {
-			this.media = IllustratedMessage.MEDIA.SCENE;
+			newMedia = IllustratedMessage.MEDIA.SCENE;
+		}
+
+		if (!(this._lastKnownBCRForMedia[newMedia] && this._shallowEqual(bcr, this._lastKnownBCRForMedia[newMedia]))) {
+			this.media = newMedia;
+			this._storeLastKnownBCRForMedia(bcr, newMedia);
+		}
+	}
+
+	_storeLastKnownBCRForMedia(bcr, media) {
+		this._lastKnownBCRForMedia[media] = { ...bcr };
+	}
+
+	_shallowEqual(object1, object2) {
```

Comment:
I suggest extracting this in packages/base/src/util as it's a common function, looks like a base one, that other components also can make use of - f.e. call it compareObjects. Other names such as shallowEqual, shalowCompare.. also sound fine.
 
 After extracting, import it
```js
import compareObjects from "@ui5/webcomponents-base/dist/compareObjects.js"
 ```
 
And, use it
 ```js
 compareObjects(objA ,objB)
 ```
 In it's on file you can add some basic JSDoc to describe the method - what it does, the params, the return value, etc.